src/helpers/redux/sagas/offline.ts

```diff
@@ -0,0 +1,37 @@
+import NetInfo from '@react-native-community/netinfo';
+import i18n from 'assets/i18n';
+import { showMessage } from 'react-native-flash-message';
+import { OFFLINE, ONLINE } from 'redux-offline-queue';
+import { eventChannel } from 'redux-saga';
+import { put, take } from 'redux-saga/effects';
+
+export function* startWatchingNetworkConnectivity(): any {
+    const channel = eventChannel((emitter) => {
+        const unsubscribe = NetInfo.addEventListener(emitter);
+        return () => unsubscribe();
+    });
+
+    try {
+        for (;;) {
+            const netStatus = yield take(channel);
+
+            if (netStatus.isConnected) {
+                yield put({ type: ONLINE });
+                showMessage({
+                    message: i18n.t('sagas.messages.yourNetworkisOnline'),
+                    type: 'success',
+                });
+            } else {
+                yield put({ type: OFFLINE });
+                showMessage({
+                    message: i18n.t('sagas.messages.yourNetworkisOffline'),
+                    type: 'danger',
+                    autoHide: !netStatus.isConnected,
+                    duration: 500000,
```

Comment:
In the picture, there's a warning at the button. We should also be careful about these warnings.
Could the reasons for that warning be this duration? Why is this so long?

Reply:
yes, that's the reason. I set this long because in order to test I need to turn the wifi off and then wasn't able to be fast enough to see the toast. Fixed.